jazz_deployments/test/test.js

```diff
@@ -1,11 +1,1181 @@
+          return obj.callback(null, responseObj, responseObj.body)
+        }
+      });
+      message = '{"errorType":"NotFound","message":"Unable to re-build mag!c as requested service is unavailable."}'
+      index.handler(event, context,(err, res) => {
+        err.should.be.equal(message);
+        sinon.assert.calledThrice(reqStub);
+        AWS.restore("DynamoDB.DocumentClient");
+        reqStub.restore();
+        return res;
+      });
+    });
+  
+    it("should indicate internal server error while initiating re-build deployment using POST method",() => {
+      event.method = "POST";
+      var responseObj = {
+        statusCode: 200,
+        body: {
+          data: {
+            token: "zaqwsxcderfv.qawsedrftg.qxderfvbhy"
+          }
+        }
+      };
+      AWS.mock("DynamoDB.DocumentClient", "query", (param, cb) => {
+        var dataObj = {
+          Items: [event.body]
+        }
+        return cb(null, dataObj);
+      });
+      reqStub = sinon.stub(request, "Request", (obj) => {
+        if(obj.method.toLowerCase() === 'post') {
+          return obj.callback(null, responseObj, responseObj.body)
+        } else if (obj.method === 'get') {
+          responseObj.body = "{\"data\":{\"errorType\":\"NotFound\"},\"input\":{\"service_id\":\"k!ngd0m_0f_mewni\"}}"
+          return obj.callback(err, null, null)
+        }
+      });
+      message = '{"errorType":"InternalServerError","message":"unhandled error occurred"}'
+      index.handler(event, context,(err, res) => {
+        err.should.be.equal(message);
+        sinon.assert.calledOnce(reqStub);
+        AWS.restore("DynamoDB.DocumentClient");
+        reqStub.restore();
+        return res;
+      });
+    });
+  
+    it("should successfully get list of deployments with provided query params using GET method", () => {
+      event.method = "GET";
+      event.path = {};
+      AWS.mock("DynamoDB", "scan", (params, cb) => {
+        var dataObj = {
+          Items: [event.body]
+        };
+        return cb(null, dataObj);
+      });
+      index.handler(event, context, (err, res) => {
+          res.should.have.deep.property('data.deployments')
+          AWS.restore("DynamoDB");
+          return res
+      })
+    });
+  
+    it("should indicate error while fecthing list of deployments with provided query params using GET method", () => {
+      event.method = "GET";
+      event.path = {};
+      message = '{"errorType":"InternalServerError","message":"unexpected error occurred"}'
+      AWS.mock("DynamoDB", "scan", (params, cb) => {
+        return cb(err, null);
+      });
+      index.handler(event, context, (err, res) => {
+        err.should.be.equal(message);
+        AWS.restore("DynamoDB");
+        return err;
+      })
+    });
+  
+    it("should successfully get deployment with provided path param using GET method", () => {
+      event.method = "GET";
```

Comment:
if you want you can add more describes here and make it simpler.
One each for post, get and delete. I see the setups are being repeated for each one of them and it can be done once(using beforeEach and afterEach) and group all these HTTP verbs in their own respective describes. What do you think?

Reply:
Worked as mentioned